src/index/coinstatsindex.cpp

```diff
@@ -0,0 +1,421 @@
+// Copyright (c) 2020-2021 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <chainparams.h>
+#include <coins.h>
+#include <crypto/muhash.h>
+#include <index/coinstatsindex.h>
+#include <serialize.h>
+#include <txdb.h>
+#include <undo.h>
+#include <validation.h>
+
+constexpr char DB_BLOCK_HASH = 's';
+constexpr char DB_BLOCK_HEIGHT = 't';
+constexpr char DB_MUHASH = 'M';
+
+namespace {
+
+struct DBVal {
+    uint256 muhash;
+    uint64_t transaction_output_count;
+    uint64_t bogo_size;
+    CAmount total_amount;
+    CAmount total_subsidy;
+    CAmount block_unspendable_amount;
+    CAmount block_prevout_spent_amount;
+    CAmount block_new_outputs_ex_coinbase_amount;
+    CAmount block_coinbase_amount;
+    CAmount unspendables_genesis_block;
+    CAmount unspendables_bip30;
+    CAmount unspendables_op_return;
+    CAmount unspendables_unclaimed_rewards;
+
+    SERIALIZE_METHODS(DBVal, obj)
+    {
+        READWRITE(obj.muhash);
+        READWRITE(obj.transaction_output_count);
+        READWRITE(obj.bogo_size);
+        READWRITE(obj.total_amount);
+        READWRITE(obj.total_subsidy);
+        READWRITE(obj.block_unspendable_amount);
+        READWRITE(obj.block_prevout_spent_amount);
+        READWRITE(obj.block_new_outputs_ex_coinbase_amount);
+        READWRITE(obj.block_coinbase_amount);
+        READWRITE(obj.unspendables_genesis_block);
+        READWRITE(obj.unspendables_bip30);
+        READWRITE(obj.unspendables_op_return);
+        READWRITE(obj.unspendables_unclaimed_rewards);
+    }
+};
+
+struct DBHeightKey {
+    int height;
+
+    DBHeightKey() : height(0) {}
```

Comment:
clang is unhappy: `unused member function 'DBHeightKey' [-Wunused-member-function]`

Reply:
fixed